### PR TITLE
Document the plugin development Vagrant file in the plugin writer's g…

### PR DESCRIPTION
**Pull request by ammaritiz**
Added quickstart docs for using Pulp Vagrant boxes for development.

closes #2909

### CODE REVIEW

File: docs/plugins/plugin-writer/developer-setup.rst [hunk 1]
```diff
@@ -0,0 +1,110 @@
+Developer Setup
+===============
+
+Each release of ``pulpcore`` on PyPI is released as a Vagrant box on `vagrantcloud.com <https://app.vagrantup.com/pulp/boxes/pulpcore>`_.
+To get started, you need to install ``Vagrant``. You can refer `this <https://docs.pulpproject.org/dev-guide/contributing/dev_setup.html#vagrant>`_.
```
Comment (dparalen) [inline docs/plugins/plugin-writer/developer-setup.rst]: - s/``Vagrant``/Vagrant/
- s/this/developer installation instructions/

File: docs/plugins/plugin-writer/developer-setup.rst [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+For backends, you can opt for either `libvirt <https://docs.pulpproject.org/dev-guide/contributing/dev_setup.html#prerequisites-for-libvirt>`_ or
```
Comment (dparalen) [inline docs/plugins/plugin-writer/developer-setup.rst]: trailing whitespace

File: docs/plugins/plugin-writer/developer-setup.rst [hunk 3]
```diff
@@ -0,0 +1,110 @@
+Developer Setup
+===============
+
+Each release of ``pulpcore`` on PyPI is released as a Vagrant box on `vagrantcloud.com <https://app.vagrantup.com/pulp/boxes/pulpcore>`_.
+To get started, you need to install ``Vagrant``. You can refer `this <https://docs.pulpproject.org/dev-guide/contributing/dev_setup.html#vagrant>`_.
+For backends, you can opt for either `libvirt <https://docs.pulpproject.org/dev-guide/contributing/dev_setup.html#prerequisites-for-libvirt>`_ or 
+`Virtualbox <https://www.virtualbox.org/wiki/Downloads>`_.
+
+Create a ``Vagrantfile`` with below configuration in an empty project directory::
```
Comment (dparalen) [inline docs/plugins/plugin-writer/developer-setup.rst]: please, put the "code" block into a file ``pulp/docs/plugins/plugin-writer/Vagrantfile`` and link it here to be downloadable:

```rest
One can use the :download:`Vagrantfile <Vagrantfile>` as a starting point and modify it as necessary.

```

File: docs/plugins/plugin-writer/developer-setup.rst [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+Start up the ``pulpcore`` Vagrant box::
+
+  $ vagrant up
+
+Once the box is downloaded and booted an ssh connection can be established with a forward tunnel for port 8000::
+
+  $ vagrant ssh -- -L 8000:localhost:8000
+
+Become user ``pulp``, activate the ``pulpenv`` virtual env, and start the webserver::
```
Comment (dparalen) [inline docs/plugins/plugin-writer/developer-setup.rst]: the log-in part is missing, I guess:

```rest
Log-in, become user ``pulp``,...

  $ vagrant ssh
  (inside the vagrant box)$ sudo su - pulp 
  ...

```

Comment (ammaritiz) [reply docs/plugins/plugin-writer/developer-setup.rst]: The user is establishing an ssh connection in the previous step. So, the log-in part is already covered.

File: docs/plugins/plugin-writer/developer-setup.rst [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+  $ sudo su - pulp
+  $ source pulpvenv/bin/activate
```
Comment (dralley) [inline docs/plugins/plugin-writer/developer-setup.rst]: ```workon pulp``` is much more ergonomic

**Pull request closed without merging**